scapy/packet.py

```diff
@@ -881,6 +881,8 @@ def clone_with(self, payload=None, **kargs):
         return pkt
 
     def __iter__(self):
+        if self.__iterlen__() == 1:
```

Comment:
Can you explain that?

Reply:
I am trying to work around this issue:
```

>>> a = IP()
>>> b = IP()
>>> pkts = [a, b]
>>> gen = SetGen(pkts)
>>> for i in gen:
...:     i.sent_time = 1
...:
>>> a.sent_time
>>> b.sent_time
>>>
```
In a fancier way that what was done before (storing all the `sent_time` values in an array then set it with a loop)

Reply:
After looking for a bit, it seems current version is the easier to setup